pkg/gateway/multiparts/tracker.go

```diff
@@ -38,21 +64,28 @@ var (
 	ErrInvalidMetadataSrcFormat = errors.New("invalid metadata source format")
 )
 
-func NewTracker(adb db.Database) Tracker {
+func NewTracker(ms kv.StoreMessage) Tracker {
```

Comment:
The name is generic, but it returns an implementation of the `Tracker`, no?

Reply:
This is the original name of the tracker and when we will be at feature complete this will also be the only implementation

Reply:
Still think that `NewKVTracker` is a better name, but not blocking..